Modules/@babylonjs/react-native/windows/BabylonReactNative/EngineViewManager.cpp

```diff
@@ -0,0 +1,148 @@
+#include "pch.h"
+#include "EngineViewManager.h"
+
+#include "JSValueReader.h"
+#include "JSValueXaml.h"
+
+using namespace winrt::Microsoft::ReactNative;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Xaml::Media;
+using namespace winrt::Windows::UI::Xaml::Controls;
+using namespace winrt::Windows::UI::Xaml::Input;
+
+namespace winrt::BabylonReactNative::implementation {
+    EngineViewManager::EngineViewManager() {}
+
+    // IViewManager
+    hstring EngineViewManager::Name() noexcept {
+        return L"EngineView";
+    }
+
+    FrameworkElement EngineViewManager::CreateView() noexcept {
+        _swapChainPanel = SwapChainPanel();

```

Comment:
I would sublcass `SwapChainPanel` and put most of this logic in that class, then have `EngineViewManager::CreateView` just be a super simple factory function. This is how it works on the other platforms. If you did that, then these revoker tokens would not be necessary as the event handler (the `SwapChainPanel` subclass) would have the same lifetime as the event source (the `SwapChainPanel` subclass). You'd still have to manage the `event_token` returned from `CompositionTarget::Rendering`, but you could do that in Loaded/Unloaded events of the `SwapChainPanel` sublcass.

Reply:
content has been moved to a new class